### ISSUE TITLE
Deprecation warnings: Using or importing the ABCs from 'collections'

### ISSUE THREAD

**Issue by brawner** [colcon/colcon-core#134]
For the Python 3.7 users these warnings pop up in many places in ROS2. They'll need to be addressed in order for compatibility with Python 3.8 (currently to be released in late 2019).

```
__________________ ERROR collecting test/test_environment.py __________________
test\test_environment.py:8: in <module>
    from colcon_core.environment import create_environment_hooks
colcon_core\environment\__init__.py:4: in <module>
    from collections import Iterable
<frozen importlib._bootstrap>:1032: in _handle_fromlist
    ???
c:\python37\lib\collections\__init__.py:52: in __getattr__
    DeprecationWarning, stacklevel=2)
E   DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
_______________ ERROR collecting test/test_environment_path.py ________________
test\test_environment_path.py:7: in <module>
    from colcon_core.environment.path import PathEnvironment
colcon_core\environment\__init__.py:4: in <module>
    from collections import Iterable
<frozen importlib._bootstrap>:1032: in _handle_fromlist
    ???
c:\python37\lib\collections\__init__.py:52: in __getattr__
    DeprecationWarning, stacklevel=2)
E   DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
____________ ERROR collecting test/test_environment_pythonpath.py _____________
test\test_environment_pythonpath.py:8: in <module>
    from colcon_core.environment.pythonpath import PythonPathEnvironment
colcon_core\environment\__init__.py:4: in <module>
    from collections import Iterable
<frozen importlib._bootstrap>:1032: in _handle_fromlist
    ???
c:\python37\lib\collections\__init__.py:52: in __getattr__
    DeprecationWarning, stacklevel=2)
E   DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
- generated xml file: C:\Users\brawner\ros2_core\build\colcon-core\pytest.xml -
```

**Comment by brawner** [comment]
I should also point out, these cause test failures when running tests with Python 3.7.

**Comment by dirk-thomas** [comment]
> They'll need to be addressed in order for compatibility with Python 3.8

Please consider to provide a pull request for the proposed change.

> these cause test failures when running tests with Python 3.7.

Can you please provide reproducible steps for this case? I ran `test` on a few packages without noticing any issues.

**Comment by brawner** [comment]
I ran colcon test on Windows 10 with python 3.7.0 using the command:

```
python install\Scripts\colcon-script.py test --merge-install --packages-select colcon-core
```

**Comment by dirk-thomas** [comment]
Addressed by #134.
